protocols/kad/src/behaviour.rs

```diff
@@ -2885,18 +2904,14 @@ pub enum QueryInfo {
         context: PutRecordContext,
     },
 
-    /// A query initiated by [`Kademlia::get_record`].
+    /// A (repeated) query initiated by [`Kademlia::get_record`].
     GetRecord {
         /// The key to look for.
         key: record::Key,
-        /// The records with the id of the peer that returned them. `None` when
-        /// the record was found in the local store.
-        records: Vec<PeerRecord>,
-        /// The number of records to look for.
-        quorum: NonZeroUsize,
-        /// The peers closest to the `key` that were queried but did not return a record,
-        /// i.e. the peers that are candidates for caching the record.
-        cache_candidates: BTreeMap<kbucket::Distance, PeerId>,
+        /// Current index of events.
+        count: usize,
```

Comment:
In my eyes tracking this as a `usize` is error prone and non-descriptive. Why not use `ProgressStep` here?

Done via cc1b2bdfc in case you want to cherry-pick.

Reply:
fixed